weasyl/controllers/content.py

```diff
@@ -502,36 +553,41 @@ def edit_submission_get_(request):
 @login_required
 @token_checked
 def edit_submission_post_(request):
-    form = request.web_input(submitid="", title="", folderid="", subtype="", rating="",
-                             content="", friends="", critique="", embedlink="")
-
-    rating = ratings.CODE_MAP.get(define.get_int(form.rating))
+    rating = ratings.CODE_MAP.get(define.get_int(request.params.get('rating', '')))
     if not rating:
         raise WeasylError("ratingInvalid")
 
     s = orm.Submission()
-    s.submitid = define.get_int(form.submitid)
-    s.title = form.title
+    s.submitid = define.get_int(request.params.get('submitid', ''))
+    s.title = request.params.get('title', '')
     s.rating = rating
-    s.content = form.content
-    s.folderid = define.get_int(form.folderid) or None
-    s.subtype = define.get_int(form.subtype)
-
-    submission.edit(request.userid, s, embedlink=form.embedlink,
-                    friends_only=form.friends, critique=form.critique)
+    s.content = request.params.get('content', '')
+    s.folderid = define.get_int(request.params.get('folderid', '')) or None
+    s.subtype = define.get_int(request.params.get('subtype', ''))
+
+    submission.edit(
+        request.userid,
+        s,
+        embedlink=request.params.get('embedlink', ''),
+        friends_only=request.params.get('friends', False),
+        critique=request.params.get('critique', False)
+    )
     raise HTTPSeeOther(location="/submission/%i/%s%s" % (
-        define.get_int(form.submitid),
-        slug_for(form.title),
+        define.get_int(s.submitid),
+        slug_for(s.title),
         "?anyway=true" if request.userid in staff.MODS else ''
     ))
 
 
 @login_required
 def edit_character_get_(request):
-    form = request.web_input(charid="", anyway="")
-    form.charid = define.get_int(form.charid)
-
-    detail = character.select_view(request.userid, form.charid, ratings.EXPLICIT.code, False, anyway=form.anyway)
+    detail = character.select_view(
+        request.userid,
+        define.get_int(request.params.get('charid')),
+        ratings.EXPLICIT.code,
+        False,
+        anyway=request.params.get('anyway', False)
```

Comment:
In general, `request.params.get('some_boolean', False)` should stay `request.params.get('some_boolean', '')` if we’re not fixing it up more, because adding `False` increases the number of representations for the same value.

Applies to a few more lines, and you’ve already done it `request.params.get('anyway', '')` in some places.

Reply:
Think I got these all cleaned up